src/interval.rs

```diff
@@ -4,20 +4,44 @@ use {Sleep, TimerError};
 
 use std::time::Duration;
 
-/// A stream representing notifications at fixed interval
+use rand::{thread_rng, Rng};
+
+/// A stream representing notifications at given interval
 ///
 /// Intervals are created through `Timer::interval`.
 #[derive(Debug)]
 pub struct Interval {
     sleep: Sleep,
-    duration: Duration,
+    min_duration: Duration,
+    max_duration: Duration,
 }
 
 /// Create a new interval
-pub fn new(sleep: Sleep, dur: Duration) -> Interval {
+pub fn new(sleep: Sleep, min_dur: Duration, max_dur: Duration) -> Interval {
     Interval {
         sleep: sleep,
-        duration: dur,
+        min_duration: min_dur,
+        max_duration: max_dur,
+    }
+}
+
+/// Returns the next duration for an interval
+/// If `min` and `max` are equal, the duration is fixed.
+/// If `min` and `max` are not equal, a duration in the range [`min`, `max`] is returned.
+///
+/// # Panics
+///
+/// Panics if `max < min`.
+pub(crate) fn next_duration(min: Duration, max: Duration) -> Duration {
+    if min == max {
+        min
+    } else {
+        let mut rng = thread_rng();
+
+        let secs = rng.gen_range(min.as_secs(), max.as_secs() + 1);
+        let nsecs = rng.gen_range(min.subsec_nanos(), max.subsec_nanos() + 1);
```

Comment:
This doesn't look right. For example if `min = Duration::from_millis(1800), max = Duration::from_millis(2900)` it will never generate `Duration::from_millis(1950)` or `Duration::from_millis(2100)`

Reply:
Yeah I see that there is a logic bug, but I don't understand that the values you are given, can not be generated? Even with the bug, these values should be generated.

Reply:
Thanks for making me aware of this bug :)